src/bokeh/palettes.py

```diff
@@ -1639,6 +1639,43 @@ def set_alpha(rgba: RGB, i: int) -> RGB:
 
     return palette
 
+def interp_palette(palette: Palette, n: int) -> Palette:
+    """ Generate a new palette by interpolating a given palette.
+
+    Linear interpolation is performed separately on each of the RGBA
+    components.
+
+    Args:
+        palette (seq[str]) :
```

Comment:
`Pallete` is defined as `Tuple[str, ...]`, so this an the return type are inconsistent with the actual types. Though I'm not sure what are the expected semantics here. Maybe this isn't expected to match types exactly?

Reply:
All the docstring mentions of `Palette` in this file use `seq[str]`, so I have gone for consistency with that. I assume the `seq[str]` was used a long time before the explicit typing of `Tuple[str, ...]`, so maybe we have tightened up the type annotations without updating docstrings. I often use `list[str]` for Bokeh palettes and don't recall this ever causing a problem, so `seq[str]` is appropriate for incoming function arguments, but perhaps the returns should be more precise as `tuple[str]`.

There are about 12 uses of `seq[str]` for returns in this file. I'll change them all but leave the arguments as they are for the moment; I can do more if desired.

Reply:
I agree that Bokeh should be permissive in what it accepts as palettes (any sequence will do) but strict it what it offers, itself (only tuples returned or presented in `bokeh.palettes`)

Reply:
OK, I've undone all changes in `palette.py` except for those to the new `interp_palette`. The docstring is definitely correct for this function, it accepts a `seq[str]` and returns a `tuple[str]`. I haven't modified any of the other functions at all, their docstrings may or may not be wrong and they may or may not accept `seq[Color]` rather than `seq[str]`. Checking that and fixing it if necessary can be done in another PR.